libs/elasticsearch-nio/src/main/java/org/elasticsearch/nio/NioSocketChannel.java

```diff
@@ -48,20 +46,36 @@ public NioSocketChannel(SocketChannel socketChannel, SocketSelector selector) th
     @Override
     public void closeFromSelector() throws IOException {
         assert socketSelector.isOnCurrentThread() : "Should only call from selector thread";
-        // Even if the channel has already been closed we will clear any pending write operations just in case
-        if (writeContext.hasQueuedWriteOps()) {
-            writeContext.clearQueuedWriteOps(new ClosedChannelException());
+
+        IOException contextCloseException = null;
+        try {
+            context.closeFromSelector();
```

Comment:
you could use `IOUtils.close(context::closeFromSelector, super::closeFromSelector)` maybe?

Reply:
`IOUtils` is in Lucene which is not a dependency. But I'll look at cleaning this up a big with `ExceptionsHelper`

Reply:
makes sense!